tests/test_om_prior.py

```diff
@@ -1,32 +1,175 @@
+@pytest.fixture(scope="session")
+def input_files(root_dir) :
+    download_input_files(root_path=root_dir,
+                         downloads=downloads,
+                         remote=remote)
+
+    input_folder = os.path.join(root_dir, "inputs")
+
+    downloaded_files = os.listdir(input_folder)
+
+    yield downloaded_files
+
+    for file in [i for i in downloaded_files if i != 'README.md'] :
+        filepath = os.path.join(input_folder, file)
+        os.remove(filepath)
+
+
+def test_001_response_for_download_links() :
+    for filename, filepath in downloads :
+        url = f"{remote}{filename}"
+        with requests.get(url, stream=True) as response :
+            print(f"Response code for {url}: {response.status_code}")
+            assert response.status_code == 200
+
+
+def test_002_cdsapi_connection(root_dir, tmpdir) :
+    filepath = tmpdir.mkdir("sub").join("test_download_cdsapi.nc")
+
+    startDate = datetime.datetime.strptime("2022-07-01", "%Y-%m-%d")
+    endDate = datetime.datetime.strptime("2022-07-02", "%Y-%m-%d")
 
+    downloadGFAS(startDate=startDate, endDate=endDate, fileName=filepath)
 
+    assert os.path.exists(filepath)
 
-def test_full_process(num_regression, root_dir, monkeypatch):
+    os.remove(filepath)
+
+
+def test_003_inputs_folder_is_empty(root_dir) :
+    input_folder = os.path.join(root_dir, "inputs")
+
+    EXPECTED_FILES = ['README.md']
+
+    assert os.listdir(input_folder) == EXPECTED_FILES, f"Folder '{input_folder}' is not empty"
+
+
+def test_004_omDownloadInputs(root_dir, input_files) :
+    EXPECTED_FILES = [
+        "ch4-electricity.csv",
+        "coal-mining_emissions-sources.csv",
+        "oil-and-gas-production-and-transport_emissions-sources.csv",
+        "NLUM_ALUMV8_250m_2015_16_alb.tif",
+        "ch4-sectoral-emissions.csv",
+        "landuse-sector-map.csv",
+        "nasa-nighttime-lights.tiff",
+        "AUS_2021_AUST_SHP_GDA2020.zip",
+        "EntericFermentation.nc",
+        "termite_emissions_2010-2016.nc",
+        "DLEM_totflux_CRU_diagnostic.nc",
+        'README.md',
+    ]
+
+    assert sorted(input_files) == sorted(EXPECTED_FILES)
+
+
+def test_005_agriculture_emissions(root_dir, input_files) :
+    filepath_livestock = os.path.join(root_dir, livestockDataPath)
+    livestock_data = xr.open_dataset(filepath_livestock)
+
+    filepath_sector = os.path.join(root_dir, sectoralEmissionsPath)
+    sector_data = pd.read_csv(filepath_sector).to_dict(orient="records")[0]
+
+    lsVal = round(np.sum(livestock_data["CH4_total"].values))
+    agVal = round(sector_data["agriculture"] * 1e9)
+    agDX = agVal - lsVal
+
+    assert agDX > 0, f"Livestock CH4 exceeds bounds of total agriculture CH4: {agDX / 1e9}"
+
+
+# TODO Update this test when file structure is clear.
+# This test ensures that the grid size for all input files is 10 km.
+def test_006_grid_size_for_geo_files(root_dir, monkeypatch) :
+    expected_cell_size = 10000
```

Comment:
I am not sure what exactly to test here. Right now the test ensure that the grid size for `geo_em.d01.nc`, `GRIDCRO2D_1`, `GRIDDOT2D_1` is 10 km.

Reply:
My initial thought was lat/lons, but then I realised that the GRIDCRO/GRIDDOT are a subset of the WRF domain.

That could be a test for after `omCreateDomainInfo.py` is run, but not applicable here